load_issues.py

```diff
@@ -0,0 +1,178 @@
+from github.Repository import Repository
+
+from loguru import logger
+
+DEFAULT_CONFIG = {"owner": "uw-ssec", "repo": "codeuw", "repos": []}
+
+
+def load_config(config_file: str = ".codeuw-config.yml") -> Dict[str, Any]:
+    """
+    Load configuration file
+
+    Parameters
+    ----------
+    config_file : str, optional
+        The string path to configuration file,
+        by default ".codeuw-config.yml"
+
+    Returns
+    -------
+    Dict[str, Any]
+        The configuration dictionary
+    """
+    config_path = Path(config_file)
+    config = yaml.safe_load(config_path.read_text())
+
+    # Set defaults, in case they're not in the config file
+    for key, value in DEFAULT_CONFIG.items():
+        config.setdefault(key, value)
+
+    return config
+
+
+def setup_github() -> Github:
+    """
+    Setup Github API
+
+    Returns
+    -------
+    Github
+        Github API object
+    """
+    from github import Auth
+
+    # using an access token
+    # it'll look for environment variable GITHUB_TOKEN
+    auth = Auth.Token(os.environ.get("GITHUB_TOKEN"))
+
+    # Public Web Github
+    return Github(auth=auth)
+
+
+def parse_issue_template(source_repo: Repository) -> Dict[str, Any]:
+    """
+    Parse the issue template from the source repository
+    and output a dictionary of the template
+
+    Parameters
+    ----------
+    source_repo : Repository
+        The source repository to parse the issue template from
+
+    Returns
+    -------
+    dict
+        The dictionary of the issue template
+    """
+    issue_template = source_repo.get_contents(path=".github/ISSUE_TEMPLATE/task.yml")
+    task_template = yaml.safe_load(issue_template.decoded_content)
+
+    # Create the body markdown template
+    body_template = ""
+    for input in task_template["body"]:
+        input_id = input["id"]
+        section_string = textwrap.dedent(
+            f'### {input["attributes"]["label"]}\n\n' f"{{{input_id}}}\n\n"
+        )
+        body_template += section_string
+
+    # Setup the template dictionary
+    template_dict = {
+        "title": (task_template["title"] + "{project_name} - {title_text}").format,
+        "labels": task_template["labels"],
+        "body": body_template.format,
+    }
+
+    return template_dict
+
+
+def generate_code_uw_issues(
+    issues_with_label: List[Issue],
+    template_dict: Dict[str, Any],
+    repo: Dict[str, Any],
+    source_issue_titles: List[str],
+    gh_repo: Repository,
+    source_repo: Repository,
+    dry_run: bool = False,
+) -> None:
+    # Loop over issues and create one by one
+    for issue in issues_with_label:
+        issue_template = template_dict.copy()
+        issue_title = issue_template["title"](
+            project_name=repo["name"], title_text=issue.title
+        )
+
+        # Skip the rest if issue already exists
+        if issue_title in source_issue_titles:
+            logger.info(f"Issue already exists in source: {issue_title}")
```

Comment:
Does this mean that if the title changes on either side this will create a duplicate issue on CodeUW? Anything we could do to prevent this? Can we check against the issue link instead?